### PR TITLE
Add a transform method for schema

### PR DESCRIPTION
**Pull request by Roms1383**
Add a transform option to convert `route.schema` on documentation generation in `dynamic` mode.

### CODE REVIEW

**Comment by Roms1383** [comment]
As a side note, I had the same errors locally as in **travis-ci** when running TypeScript `tsc --project ./tsconfig.json` the first time after install : I had to install `@types/node` to fix it.

**Comment by mcollina** [comment]
I’m not sure why, but can you add that as a devDependency?

We’ll need CI to pass to land this.

**Comment by Roms1383** [comment]
Done @mcollina, waiting for CI

**Comment by Roms1383** [comment]
Ah, forgot the typings sorry...

File: index.d.ts [hunk 1]
```diff
@@ -20,6 +20,11 @@ declare namespace fastifySwagger {
   interface FastifyDynamicSwaggerOptions extends FastifySwaggerOptions {
     mode?: 'dynamic';
     swagger?: Partial<SwaggerSchema.Spec>;
+    /**
+     * Overwrite the route schema
+     * @default false
+     */
```
Comment (mcollina) [inline index.d.ts]: I don’t think the default is false. Typings also have tests.

Comment (Roms1383) [reply index.d.ts]: Yeah, sorry I'm not a heavy user of TypeScript, I'm gonna fix it and currently looking at for the unit-tests

Comment (Roms1383) [reply index.d.ts]: Where do you usually test dynamic mode ?

Comment (Roms1383) [reply index.d.ts]: Sorry for being so long about the unit-tests, I'm not familiar with `tap`.
Also I don't know to which extend you need to test the property.

**Comment by Roms1383** [comment]
Ok it should be good by now :)
I had to add the devDependencies for the unit-tests : if you think that's not ok,  please provide me with feedback.

**Comment by Roms1383** [comment]
Anyway my guess is that the user who implements `transform` method has the responsibility to make it work and can even filter which one to transform or not so that shouldn't be blocking.

**Comment by mcollina** [comment]
This seems a pretty innocent feature to have. I don't really understand your points, would you mind opening an issue or a follow up PR?

**Comment by Roms1383** [comment]
@mcollina Actually after testing and looking at documentation, to make the story short that's OK 😃 

In details, my test was something like :
```js
const params = ...
const body = ...
const querystring = ...
const response = {
  200: Joi.object().keys({ message: Joi.string().required() }).required()
}
```
Looking at **fastify** documentation I understand why adding a **joi** object for `schema`'s `response` generate an issue : as stated [here](https://github.com/fastify/fastify/blob/master/docs/Validation-and-Serialization.md#serialization) the `response` will anyway get used with **fast-json-stringify** which is the bug I got 👍 
```shell
not ok Converting circular structure to JSON
    found:
      name: TypeError
      stack: |-
        TypeError: Converting circular structure to JSON
            at stringify (/Users/romain/Development/sandbox/fastify-swagger/node_modules/fast-json-stable-stringify/index.js:42:19)
            at stringify (/Users/romain/Development/sandbox/fastify-swagger/node_modules/fast-json-stable-stringify/index.js:50:25)
            at stringify (/Users/romain/Development/sandbox/fastify-swagger/node_modules/fast-json-stable-stringify/index.js:50:25)
            at module.exports (/Users/romain/Development/sandbox/fastify-swagger/node_modules/fast-json-stable-stringify/index.js:58:7)
            at Ajv._addSchema (/Users/romain/Development/sandbox/fastify-swagger/node_modules/ajv/lib/ajv.js:293:30)
            at Ajv.compile (/Users/romain/Development/sandbox/fastify-swagger/node_modules/ajv/lib/ajv.js:111:24)
            at isValidSchema (/Users/romain/Development/sandbox/fastify-swagger/node_modules/fast-json-stringify/index.js:917:7)
            at build (/Users/romain/Development/sandbox/fastify-swagger/node_modules/fast-json-stringify/index.js:39:3)
            at getValidatorForStatusCodeSchema (/Users/romain/Development/sandbox/fastify-swagger/node_modules/fastify/lib/validation.js:13:10)
            at /Users/romain/Development/sandbox/fastify-swagger/node_modules/fastify/lib/validation.js:19:21
            at Array.reduce (<anonymous>)
            at getResponseSchema (/Users/romain/Development/sandbox/fastify-swagger/node_modules/fastify/lib/validation.js:18:22)
            at build (/Users/romain/Development/sandbox/fastify-swagger/node_modules/fastify/lib/validation.js:53:31)
            at afterRouteAdded (/Users/romain/Development/sandbox/fastify-swagger/node_modules/fastify/fastify.js:636:9)
            at /Users/romain/Development/sandbox/fastify-swagger/node_modules/fastify/fastify.js:596:7
            at Object._encapsulateThreeParam (/Users/romain/Development/sandbox/fastify-swagger/node_modules/avvio/boot.js:389:7)
      message: Converting circular structure to JSON
    at:
      line: 62
      column: 7
      file: test/transform.js
      function: fastify.ready.err
    stack: |
      fastify.ready.err (test/transform.js:62:7)
      Object._encapsulateThreeParam (node_modules/avvio/boot.js:380:13)
      Boot.callWithCbOrNextTick (node_modules/avvio/boot.js:311:5)
      release (node_modules/fastq/queue.js:127:16)
      Object.resume (node_modules/fastq/queue.js:61:7)
      Boot.Plugin.loadPlugin.call (node_modules/avvio/boot.js:146:20)
      toLoad.finish (node_modules/avvio/plugin.js:169:7)
      done (node_modules/avvio/plugin.js:127:5)
      check (node_modules/avvio/plugin.js:138:7)
    source: |
      t.error(err)
```
Last but not least it's actually mentioned [here](https://github.com/fastify/fastify/blob/master/docs/Validation-and-Serialization.md#schema-compiler) that :
> (...) maybe you want to change the validation library. Perhaps you like Joi. In this case, you can use it to validate the url parameters, body, and query string!

And there's no mention of validating `schema`'s `response`, so it makes sense.
Thanks !

**Pull request merged**